lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -2268,12 +2272,98 @@ static void computeReuse(TransferReadOp readOp, VectState *state) {
   }
 }
 
-// Compute the reuse interval for all the transfer_read operations. The
-// transfer_read operations capture the vector load. Since AIE only allows for
-// aligned vector loads, we need to compose multiple transfer reads together to
-// form intervals of certain width (128, 256, 512, or 1024), and create an AIE
-// vector from each interval.
-static void computeReuseInFunc(func::FuncOp func, VectState *state) {
+static LogicalResult isUnalignedLoad(TransferReadOp readOp, VectState *state) {
+  VectorType vectorType = readOp.getResult().getType().cast<VectorType>();
+  unsigned lanes = getVectorLaneSize(vectorType);
+
+  AffineExpr linearAccess = constructLinearizedAffineExpr(readOp, state);
+  if (linearAccess.isSymbolicOrConstant()) {
+    return success();
+  }
+
+  MemRefType memRefType = readOp.getSource().getType().cast<MemRefType>();
+  MLIRContext *context = memRefType.getContext();
+  ArrayRef<int64_t> sizes = memRefType.getShape();
+  int numDims = sizes.size();
+
+  auto block = readOp->getBlock();
+  assert(state->blockToEnclosingLoops.count(block) &&
+         "enclosing loops should have been computed for the read operation\n");
+  auto enclosingLoops = state->blockToEnclosingLoops[block];
+
+  SmallVector<Value, 4> indices(readOp.getIndices().begin(),
+                                readOp.getIndices().end());
+
+  // If the lowest dim has iv, check whether its corresponding loop step can
+  // be divisible by the vector lanes.
+  int32_t step = 0;
+  if (auto dimExpr =
+          getAffineDimExpr(numDims - 1, context).dyn_cast<AffineDimExpr>()) {
+    auto index = indices[dimExpr.getPosition()];
+    // Iterate over all enclosing loops, and find the one that is variant in
+    // index.
+    for (auto loop : enclosingLoops) {
+      AffineForOp affineForOp = cast<AffineForOp>(loop);
+      auto iv = affineForOp.getInductionVar();
+      auto invariants = getInvariantAccesses(iv, indices);
+
+      if (!invariants.count(index)) {
+        step = affineForOp.getStep();
+        if (step % lanes) {
+          LLVM_DEBUG(llvm::dbgs() << "\n\n"
```

Comment:
Is this message printed only if the code was compiled for debug? If so, then the user will have no way to figure out what went wrong in case the the alignment check first.